orca-clouddriver/src/main/groovy/com/netflix/spinnaker/orca/clouddriver/pipeline/job/PreconfiguredJobStage.groovy

```diff
@@ -22,17 +22,21 @@ import com.netflix.spinnaker.orca.clouddriver.exception.PreconfiguredJobNotFound
 import com.netflix.spinnaker.orca.clouddriver.service.JobService
 import com.netflix.spinnaker.orca.pipeline.TaskNode
 import com.netflix.spinnaker.orca.pipeline.model.Stage
-import org.springframework.beans.factory.annotation.Autowired
 import org.springframework.stereotype.Component
 
 @Component
 class PreconfiguredJobStage extends RunJobStage {
 
-  @Autowired(required=false)
   private JobService jobService
 
+  public PreconfiguredJobStage() {}
```

Comment:
Is the no-arg constructor used?

Reply:
There used to be a `@Autowired(required=false)` on `jobService` so I figured I need it? How do you handle non-required beans with constructor injection?

Reply:
Ah, I see, good question---the way to do that is to inject an `Optional<JobService> optionalJobService` into the constructor, then have the constructor do something like `this.jobService = optionalJobService.orElse(null)` in the constructor.